File: core/src/main/java/com/ctrip/ferriswheel/core/action/EraseCells.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;

public class EraseCells extends TableAction implements Action {
    private int top;
    private int right;
    private int bottom;
    private int left;

    public EraseCells() {
    }

    public EraseCells(String sheetName, String tableName,
                      int top, int right, int bottom, int left) {
        super(sheetName, tableName);
        this.top = top;
        this.right = right;
        this.bottom = bottom;
        this.left = left;
    }

    public int getTop() {
        return top;
    }

    public void setTop(int top) {
        this.top = top;
    }

    public int getRight() {
        return right;
    }

    public void setRight(int right) {
        this.right = right;
    }

    public int getBottom() {
        return bottom;
    }

    public void setBottom(int bottom) {
        this.bottom = bottom;
    }

    public int getLeft() {
        return left;
    }

    public void setLeft(int left) {
        this.left = left;
    }
}

File: ui/react/src/action/EraseCells.ts
import TableAction from "./TableAction";
import Action from "./Action";

class EraseCells extends TableAction {
    public top: number;
    public right: number;
    public bottom: number;
    public left: number;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const top: number = input.top;
        const right: number = input.right;
        const bottom: number = input.bottom;
        const left: number = input.left;
        return new EraseCells(sheetName, tableName, top, right, bottom, left);
    }

    constructor(
        sheetName: string,
        tableName: string,
        top: number,
        right: number,
        bottom: number,
        left: number) {

        super(sheetName, tableName);
        this.top = top;
        this.right = right;
        this.bottom = bottom;
        this.left = left;
    }

    public wrapper(): Action {
        const action = new Action();
        action.eraseCells = this;
        return action;
    }

}

export default EraseCells;
